tests/tests/dlog_eq.rs

```diff
@@ -0,0 +1,91 @@
+#[generic_tests::define]
+mod interactive {
+    use generic_ec_zkp::dlog_eq::interactive as dlog_eq;
+
+    #[test]
+    fn passing_test<E: generic_ec::Curve>() {
+        let mut rng = rand_dev::DevRng::new();
+
+        let secret_share = generic_ec::SecretScalar::<E>::random(&mut rng);
+
+        let base = generic_ec::Point::generator() * generic_ec::Scalar::random(&mut rng);
+        let data = dlog_eq::Data::from_secret_key(&secret_share, base);
+
+        let (comm, pcomm) = dlog_eq::commit_data(data, &mut rng);
+        let challenge = generic_ec::Scalar::random(&mut rng);
+        let proof = dlog_eq::prove(pcomm, challenge, &secret_share);
+        dlog_eq::verify(data, comm, challenge, proof).unwrap()
+    }
+    #[test]
+    fn failing_test<E: generic_ec::Curve>() {
+        let mut rng = rand_dev::DevRng::new();
+
+        let secret_share = generic_ec::SecretScalar::<E>::random(&mut rng);
+
+        let base = generic_ec::Point::generator() * generic_ec::Scalar::random(&mut rng);
+        let mut data = dlog_eq::Data::from_secret_key(&secret_share, base);
+        // Replace the exponentiation with something wrong
+        data.exp2 += generic_ec::Point::generator();
+
+        let (comm, pcomm) = dlog_eq::commit_data(data, &mut rng);
+        let challenge = generic_ec::Scalar::random(&mut rng);
+        let proof = dlog_eq::prove(pcomm, challenge, &secret_share);
+        assert!(
+            dlog_eq::verify(data, comm, challenge, proof).is_err(),
+            "proof should fail"
+        )
+    }
+
+    #[instantiate_tests(<generic_ec::curves::Secp256k1>)]
+    mod secp256k1 {}
+    #[instantiate_tests(<generic_ec::curves::Secp256r1>)]
+    mod secp256r1 {}
+    #[instantiate_tests(<generic_ec::curves::Stark>)]
+    mod stark {}
+    #[instantiate_tests(<generic_ec::curves::Ed25519>)]
+    mod ed25519 {}
+}
+
+#[generic_tests::define]
+mod non_interactive {
+    use generic_ec_zkp::dlog_eq::non_interactive as dlog_eq;
+
+    #[test]
+    fn passing_test<E: generic_ec::Curve, D: digest::Digest>() {
+        let mut rng = rand_dev::DevRng::new();
+        let shared_state = "shared state";
+
+        let secret_share = generic_ec::SecretScalar::random(&mut rng);
+
+        let base = generic_ec::Point::generator() * generic_ec::Scalar::random(&mut rng);
+        let data = dlog_eq::Data::from_secret_key(&secret_share, base);
+
+        let proof = dlog_eq::prove::<D, E>(&shared_state, &secret_share, data, &mut rng);
+        dlog_eq::verify::<D, E>(&shared_state, data, proof).unwrap();
```

Comment:
Usually `E` generic param goes first, and then `D` follows, like as you wrote in test function:

https://github.com/LFDT-Lockness/generic-ec/blob/7fec99d254c1e06220e0cb8aa859ff8f32158466/tests/tests/dlog_eq.rs#L53-L54

Maybe change order in `non_interactive` functions? It's a good idea to keep that order consistent

Reply:
On the other hand, we usually keep a parameter that is replaced with `_` last, like [here](https://docs.rs/cggmp21/latest/cggmp21/signing/struct.SigningBuilder.html#method.set_derivation_path_with_algo). I don't care either way for this function though